lib/libzfs/libzfs_sendrecv.c

```diff
@@ -788,8 +788,8 @@ dump_ioctl(zfs_handle_t *zhp, const char *fromsnap, uint64_t fromsnap_obj,
 	libzfs_handle_t *hdl = zhp->zfs_hdl;
 	nvlist_t *thisdbg;
 
-	assert(zhp->zfs_type == ZFS_TYPE_SNAPSHOT);
-	assert(fromsnap_obj == 0 || !fromorigin);
+	ASSERT3S(zhp->zfs_type, ==, ZFS_TYPE_SNAPSHOT);
```

Comment:
It's not very clear, but we've tried to avoid using the `ASSERT` macros in userspace-only-code, and instead use `assert(3)` as would be normal.  The `ASSERT`/`VERIFY` macros are intended for code that's build in both user and kernel contexts.  Clearly we could have done a better job at that.  Functionally it of course doesn't really matter.